### PR TITLE
feat: aws lambda events serialization

### PR DESCRIPTION
**Pull request by sdelamo**
This is a PR provides [Micronaut Serialization](https://micronaut-projects.github.io/micronaut-serialization/latest/guide/) to [AWS Event Libs](https://github.com/aws/aws-lambda-java-libs). We need to support the following objects. I have copied sample JSON files from the AWS Console for the tests.

- [x] APIGatewayCustomAuthorizerEvent
- [ ] APIGatewayProxyRequestEvent
- [ ] APIGatewayProxyResponseEvent
- [ ] APIGatewayV2CustomAuthorizerEvent
- [x] APIGatewayV2HTTPEvent
- [ ] APIGatewayV2HTTPResponse
- [ ] APIGatewayV2ProxyRequestEvent
- [ ] APIGatewayV2ProxyResponseEvent
- [ ] APIGatewayV2WebSocketEvent
- [ ] APIGatewayV2WebSocketResponse
- [x] ActiveMQEvent
- [x] AppSyncLambdaAuthorizerEvent
- [x] AppSyncLambdaAuthorizerResponse
- [ ] ApplicationLoadBalancerRequestEvent
- [ ] ApplicationLoadBalancerResponseEvent
- [ ] CloudFormationCustomResourceEvent
- [ ] CloudFrontEvent
- [x] CloudWatchLogsEvent
- [x] CodeCommitEvent
- [ ] CognitoEvent
- [x] CognitoUserPoolCreateAuthChallengeEvent
- [x] CognitoUserPoolCustomMessageEvent
- [x] CognitoUserPoolDefineAuthChallengeEvent
- [ ] CognitoUserPoolEvent
- [x] CognitoUserPoolMigrateUserEvent
- [x] CognitoUserPoolPostAuthenticationEvent
- [x] CognitoUserPoolPostConfirmationEvent
- [x] CognitoUserPoolPreAuthenticationEvent
- [x] CognitoUserPoolPreSignUpEvent
- [x] CognitoUserPoolPreTokenGenerationEvent
- [x] CognitoUserPoolVerifyAuthChallengeResponseEvent
- [x] ConfigEvent
- [x] ConnectEvent
- [ ] DynamodbEvent
- [ ] DynamodbTimeWindowEvent
- [ ] IamPolicyResponse
- [ ] IamPolicyResponseV1
- [x] IoTButtonEvent
- [ ] KafkaEvent
- [x] KinesisAnalyticsFirehoseInputPreprocessingEvent
- [x] KinesisAnalyticsInputPreprocessingResponse
- [x] KinesisAnalyticsOutputDeliveryEvent
- [x] KinesisAnalyticsOutputDeliveryResponse
- [x] KinesisAnalyticsStreamsInputPreprocessingEvent
- [ ] KinesisEvent
- [x] KinesisFirehoseEvent
- [ ] KinesisTimeWindowEvent
- [ ] LambdaDestinationEvent
- [ ] LexEvent
- [ ] RabbitMQEvent
- [ ] S3BatchEvent
- [x] S3BatchResponse
- [ ] S3Event
- [ ] S3ObjectLambdaEvent
- [ ] SNSEvent
- [x] SQSBatchResponse
- [x] SQSEvent
- [x] ScheduledEvent
- [x] SecretsManagerRotationEvent
- [x] SimpleIAMPolicyResponse
- [x] StreamsEventResponse
- [x] TimeWindowEventResponse

### CODE REVIEW

**Comment by sdelamo** [comment]
@hollingsworthd can you do this?

File: aws-events-serde/src/main/java/io/micronaut/aws/lambda/events/serde/APIGatewayCustomAuthorizerEventSerde.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2017-2021 original authors
```
Comment (hollingsworthd) [inline aws-events-serde/src/main/java/io/micronaut/aws/lambda/events/serde/APIGatewayCustomAuthorizerEventSerde.java:2]: 2022?

**Review by hollingsworthd — COMMENTED** [review]
I had some minor documentation concerns, but it looks good to me

**Comment by sdelamo** [comment]
@hollingsworthd what I want you to do is continue the PR. I have only created `@SerdeImport` for some of the events. As, you see there are a lot of event which we have to create serialization configuration for.

**Comment by yashLadha** [comment]
@timyates are you actively working on this PR? If not i also faced similar issue while building a sample lambda and would like to solve this issue for others.

**Comment by sdelamo** [comment]
> @timyates are you actively working on this PR? If not i also faced similar issue while building a sample lambda and would like to solve this issue for others.

@yashLadha we are going to merge this today.

**Pull request merged**